packages/dart_frog_test/lib/src/dart_frog_test_context.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:dart_frog/dart_frog.dart';
+import 'package:mocktail/mocktail.dart';
+
+class _MockRequestContext extends Mock implements RequestContext {}
+
+/// {@template dart_frog_test_context}
+/// A context used to test Dart Frog handlers.
+/// {@endtemplate}
+class DartFrogTestContext {
+  /// {@macro dart_frog_test_context}
+  DartFrogTestContext({
+    required this.path,
+    this.basePath = 'https://test.com',
+    this.method = HttpMethod.get,
+    this.headers,
+    this.body,
+  });
+
+  /// The route path.
+  final String path;
+
+  /// The base path of the request.
+  final String basePath;
```

Comment:
Out of curiosity, why would a developer change the basePath during testing?

Reply:
That is a fair question. Honestly, I can't think on any. But we need a base path in order to do the parsing of the `Uri`, and asking the user to inform it all the time would be a little bit verbose.

So I thought that we could just put in an attribute with a default, so in case there is a context where they would want to put a different base path, there is the option.